### PR TITLE
Simple signer CLI

### PR DESCRIPTION
**Pull request by kantai**
This is a simple transaction generation and signer CLI.

For now, this only supports standard single-single transactions. We can expand the feature set later if we want it to be expanded (good multi-sig support would probably require some refactoring in the TransactionAuth, Transaction, and TransactionSigner structs, as well as breaking generation & signing into two different CLI methods).

```
$ ./target/debug/blockstack-cli -h
blockstack-cli (options) [method] [args...]

This CLI allows you to generate simple signed transactions for blockstack-core
to process.

This CLI has two methods:

  publish          used to generate and sign a contract publish transaction
  contract-call    used to generate and sign a contract-call transaction
  generate-sk      used to generate a secret key for transaction signing

For usage information on those methods, call `blockstack-cli [method] -h`

`blockstack-cli` accepts flag options as well:

   --testnet       instruct the transaction generator to use a testnet version byte instead of MAINNET (default)


$ ./target/debug/blockstack-cli generate-sk -h
USAGE:
 blockstack-cli (options) generate-sk

This method generates a secret key, outputting the hex encoding of the
secret key, the corresponding public key, and the corresponding P2PKH Stacks address.

$ ./target/debug/blockstack-cli publish -h
USAGE:
 blockstack-cli (options) publish [publisher-secret-key-hex] [fee-rate] [nonce] [contract-name] [file-name.clar]

The publish command generates and signs a contract publish transaction. If successful,
this command outputs the hex string encoding of the transaction to stdout, and exits with
code 0

$ ./target/debug/blockstack-cli contract-call -h
USAGE:
 blockstack-cli (options) contract-call [origin-secret-key-hex] [fee-rate] [nonce] [contract-publisher-address] [contract-name] [function-name] [args...]

The contract-call command generates and signs a contract-call transaction. If successful,
this command outputs the hex string encoding of the transaction to stdout, and exits with
code 0

Arguments are supplied in one of two ways: through script evaluation or via hex encoding
of the value serialization format. The method for supplying arguments is chosen by
prefacing each argument with a flag:

  -e  indicates the argument should be _evaluated_
  -x  indicates the argument is supplied as a hexstring of the value serialization

e.g.,

   blockstack-cli contract-call $secret_key SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4 10 0 foo-contract \
      transfer-fookens -e \'SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4 \
                       -e "(+ 1 2)" \
                       -x 0000000000000000000000000000000001 \
                       -x 050011deadbeef11ababffff11deadbeef11ababffff

```

Note that arguments supplied via the `-x` flag currently would want the _JSON_ serialization scheme, but would change to the more compact serialization once #1199 merges.

### CODE REVIEW

File: src/blockstack_cli.rs [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+    StacksTransaction, TransactionSmartContract, TransactionContractCall, StacksAddress };
+use blockstack_lib::burnchains::Address;
+use blockstack_lib::net::{Error as NetError, StacksMessageCodec};
+
+const USAGE: &str = "blockstack-cli (options) [method] [args...]
+
+This CLI allows you to generate simple signed transactions for blockstack-core
+to process.
+
+This CLI has two methods:
```
Comment (jcnelson) [inline src/blockstack_cli.rs]: "three methods"

File: src/blockstack_cli.rs [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+Arguments are supplied in one of two ways: through script evaluation or via hex encoding
+of the value serialization format. The method for supplying arguments is chosen by
+prefacing each argument with a flag:
+
+  -e  indicates the argument should be _evaluated_
+  -x  indicates the argument is supplied as a hexstring of the value serialization
+
+e.g.,
+
+   blockstack-cli contract-call $secret_key SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4 10 0 foo-contract \\
```
Comment (jcnelson) [inline src/blockstack_cli.rs]: This example doesn't match the usage.  What does the address `SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4` correspond to?

File: src/blockstack_cli.rs [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+      transfer-fookens -e \\'SPJT598WY1RJN792HRKRHRQYFB7RJ5ZCG6J6GEZ4 \\
+                       -e \"(+ 1 2)\" \\
+                       -x 0000000000000000000000000000000001 \\
+                       -x 050011deadbeef11ababffff11deadbeef11ababffff
```
Comment (jcnelson) [inline src/blockstack_cli.rs]: To clarify, does `-x` indicate that a serialized Clarity value is being passed?

Comment (kantai) [reply src/blockstack_cli.rs]: Yes

**Pull request merged**